### DISCUSSION TITLE
FreeRDP client over UEFI

### DISCUSSION THREAD

**Discussion by daalbano** [FreeRDP/FreeRDP#9444] [General]
Does FreeRDP support UEFI? Would be very interesting being able to assemble super simple stupid clients that boot via UEFI and use the UEFI interface for the networking, the I/O, the display and the USB redirection.

I know that FreeRDP, alone, wouldn't be enough (freeRDP is a cli client) but I am wondering if there is any support for UEFI as UEFI has posix support.

**Comment by akallabeth** [comment]
@daalbano no, currently not.
1. would require writing a new client that supports the graphics and keyboard/mouse API for `UEFI`
2. `FreeRDP` is not a `cli client` but a `RDP` client and server library (and for showcase we provide some simple client and server implementations)

**Reply by daalbano** [reply]
Thanks for the reply!

Yes, I am aware that FreeRDP is not just a client, sorry I said "is a cli client", but what I meant was that you need "something" else that will leverage FreeRDP to make things happen.

I was having a look at the SDL & the sample client, I will try to do a plain build as UEFI firmware of the sample client (that does nothing) to see where it gets and what needs to be fixed dependencies & build side.
